File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.java
class Solution {
    public int characterReplacement(String s, int k) {
        int len = s.length();
        if(len == 0){
            return 0;
        }
        int[] count = new int[26];
        int res = 1, start = 0, maxCount = 0;
        for(int end = 0 ; end < len ; end ++){
            count[s.charAt(end)  - 'A']++;
            maxCount = Math.max(maxCount, count[s.charAt(end)  - 'A']);

            while (maxCount + k < end - start + 1){
                count[s.charAt(start) - 'A']--;
                start++;
            }

            res = Math.max(res, end - start + 1);
        }
        
        return res;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-1.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        i = 0
        j = len(heights) - 1
        res = 0
        while i < j:
            curr = min(heights[j], heights[i]) * (j - i)
            res = max(res, curr)
            if heights[i] > heights[j]:
                j -= 1
            else:
                i += 1
        
        return res
        